shared/src/business/useCaseHelper/consolidatedCases/formatPublicCase.ts

```diff
@@ -0,0 +1,48 @@
+import { NotFoundError } from '../../../errors/errors';
+import { PublicCase } from '../../entities/cases/PublicCase';
+import {
+  caseContactAddressSealedFormatter,
+  caseSealedFormatter,
+} from '../../utilities/caseFilter';
+import { decorateForCaseStatus } from '../../useCases/getCaseInteractor';
+import { isSealedCase } from '../../entities/cases/Case';
+
+// TODO: WRITE UNNNNNNIIIIIIITTTTTTT TEST!!!!!!!!!!!
+
+/**
+ * formatPublicCase takes a rawCase and formats to a public case
+ *
+ * @param {object} providers the providers object
+ * @param {string} providers.applicationContext the applicationContext
+ * @param {string} providers.docketNumber the docket number of the case to get
+ * @param {object} providers.rawCaseRecord the rawCaseRecord
+ * @returns {object} the validated public case data
+ */
+export const formatPublicCase = ({
+  applicationContext,
+  docketNumber,
+  rawCaseRecord,
+}: {
+  applicationContext: IApplicationContext;
+  rawCaseRecord?: TCase;
+  docketNumber: string;
+}) => {
+  if (!rawCaseRecord.docketNumber && !rawCaseRecord.entityName) {
+    const error = new NotFoundError(`Case ${docketNumber} was not found.`);
+    error.skipLogging = true;
+    throw error;
+  }
+  rawCaseRecord.isSealed = isSealedCase(rawCaseRecord);
+  if (isSealedCase(rawCaseRecord)) {
```

Comment:
Could this if condition check for `rawCaseRecord.isSealed`, and remove the duplicative call to `isSealedCase(rawCaseRecord)`?

Reply:
This one was a bit odd. `caseSealedFormatter` grabs several properties from the `rawCaseRecord`, including `isSealed` and returns it. It is possible given the logic in `Case.js` for `isSealed` to be falsy and `sealedDate` to be truthy and we would still treat the case as sealed. I do not know if this is a state that we would ever naturally end up in, but I think the previous logic probably accounted for this. We also did not want to make changes to other areas for this bug, although I certainly would not fault someone for wanting to track down whether that is a valid scenario, and if not, correct it.

We went with this new logic to hopefully make it more clear, although a bit verbose, that we are setting `isSealed` regardless.

Reply:
This seems like a good opportunity to tighten some of the validation logic for a Case. It also seems funky to me that it can get in that state (`isSealed` === false and `sealedDate` is defined) since we have methods off the Case to seal and unseal where both values are updated set. I feel like more investigation into how this is happening is warranted, maybe we aren't using that method somewhere where we should be

Reply:
Ok we are going to time box the investigation to see how cases are being set (with or without the sealedDate reset) and perhaps create a devex task or DOD item for the bug.